java/src/main/java/io/cucumber/gherkin/GherkinTokenMatcher.java

```diff
@@ -90,71 +101,75 @@ public boolean match_Other(Token token) {
 
     @Override
     public boolean match_Empty(Token token) {
-        if (!token.line.isEmpty()) {
+        if (token.line.isEmpty()) {
+            setTokenMatched(token, TokenType.Empty, null, null, 0, null, null);
+            return true;
+        } else {
             return false;
         }
-        setTokenMatched(token, TokenType.Empty, null, null, 0, null, null);
-        return true;
     }
 
     @Override
     public boolean match_Comment(Token token) {
-        if (!token.line.startsWith(COMMENT_PREFIX_CHAR)) {
+        if (token.line.startsWith(COMMENT_PREFIX_CHAR)) {
+            String text = token.line.getRawText();
+            setTokenMatched(token, TokenType.Comment, text, null, 0, null, null);
+            return true;
+        } else {
             return false;
         }
-        String text = token.line.getRawText();
-        setTokenMatched(token, TokenType.Comment, text, null, 0, null, null);
-        return true;
     }
 
     @Override
     public boolean match_Language(Token token) {
-        if (!token.line.startsWith(COMMENT_PREFIX_CHAR)) {
-            return false;
-        }
-        Matcher matcher = LANGUAGE_PATTERN.matcher(token.line.getText());
-        if (!matcher.matches()) {
+        if (token.line.startsWith(COMMENT_PREFIX_CHAR)) {
+            Matcher matcher = LANGUAGE_PATTERN.matcher(token.line.getText());
+            if (!matcher.matches()) {
+                return false;
+            }
+            String language = matcher.group(1);
+            setTokenMatched(token, TokenType.Language, language, null, token.line.getIndent(), null, null);
+            setLanguageMatched(language, token.location);
+            return true;
+        } else {
             return false;
         }
-        String language = matcher.group(1);
-        setTokenMatched(token, TokenType.Language, language, null, token.line.getIndent(), null, null);
-        setLanguageMatched(language, token.location);
-        return true;
     }
 
     @Override
     public boolean match_TagLine(Token token) {
-        if (!token.line.startsWith(TAG_PREFIX_CHAR)) {
+        if (token.line.startsWith(TAG_PREFIX_CHAR)) {
```

Comment:
I intentionally rewrote these to early-out style. It avoids indentation, makes the code a bit easier to read.